### ISSUE TITLE
[Gnome 46 and Wayland] Windowed Fullscreen isn't fullscreen

### ISSUE THREAD

**Issue by Enderteck** [xCollateral/VulkanMod#459]
![image](https://github.com/user-attachments/assets/bf53be6b-1e95-46fb-b29e-ff06c4607424)

When Windowed fullcreen is enabled, the window doesn't take the entire screen but leaves the taskbar visible.

Minecraft : 1.21
Fabric : 0.16.0
Vulkanmod : 4.3
Fedora Workstation 40
Gnome 46
Wayland and AMD GPU

**Comment by sqrvrt** [comment]
Works as intended AKA borderless mode. Use acual fullscreen if you want fullscreen.

**Comment by Enderteck** [comment]
You sure ?

That seems kinda stupid, no other game provides this, Windowed fullscreen always allows you to have a fullscreen window while being able to have the game open when not focused.

If this isn't supposed to be the borderless Window that every single other game has, it shoud'nt be advertized as fullscreen, as it's litteraly not fullscreen.

**Comment by lv37** [comment]
It is supposed to be fullscreen. This is definetely a bug

**Comment by sqrvrt** [comment]
AFAIK that's just how gnome works. If you want fullscreen, a widow should advertise itself as fullscreen. If you want a window without a title bar, the window should advertise itself as borderless (or, rather, just not include a titlebar). From what I see the OP also uses a dash to panel extension, and they may want to report it there instead since the top bar (the default component) is correctly hidden.

I still do not understand why would you want to get a borderless fullscreen here since originally it's just a hack to bypass a dumb rule on Windows.

TL;DR Dash to panel (gnome extension) is probably at fault here instead, unless proven otherwise by tests on other mods.

**Comment by Enderteck** [comment]
No definetly a Fullscreen issue and not a Dash to Panel.

![image](https://github.com/user-attachments/assets/221995b8-1f3c-4e53-9251-bf67a584a74c)

Borderless windowed allows me to alt tab and click different apps without minimizing the game.

Every other game I have tried on Gnome distribution has had real Borderless fullscreen, native Linux games and through wine, this is a Vulkanmod on Gnome Wayland issue, but it's the default on the most popular distros.

**Comment by UleyeyeUwma** [comment]
This is an issue on KDE Plasma under Wayland too, in Plasma the issue is even more noticeable, because it doesn't even try to set itself to fullscreen, it just sets the resolution to the screen resolution, but it remains windowed, border decorations do not get removed and window position isn't changed, the window does not set itself as "Fullscreen" in the window properties.

The workaround on Plasma is to manually set the window property to fullscreen, but this is definitely not how windowed fullscreen is supposed to work.

**Comment by Versatile-V1** [comment]
Is this still an issue?

**Comment by Enderteck** [comment]
Yes

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/5c3dcfaa-452d-44c1-be49-232c84a941b1" />

**Comment by Versatile-One** [comment]
> Yes
> <img alt="Image" width="1920" height="1080" src="https://private-user-images.githubusercontent.com/123917079/526297378-5c3dcfaa-452d-44c1-be49-232c84a941b1.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3ODA0MDgxMTksIm5iZiI6MTc4MDQwNzgxOSwicGF0aCI6Ii8xMjM5MTcwNzkvNTI2Mjk3Mzc4LTVjM2RjZmFhLTQ1MmQtNDRjMS1iZTQ5LTIzMmM4NGE5NDFiMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwNjAyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDYwMlQxMzQzMzlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04ZTIxNjlhZDM0NDMzYTBkNTY3N2RjNjBhMDQ3ZjcwNTdkZTMyNDVhMzc1YmI4NDRjYmFhN2UxZDFjODc2ZGJlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZyZXNwb25zZS1jb250ZW50LXR5cGU9aW1hZ2UlMkZwbmcifQ.3Au8_at0vDrOg6K9uL1C5B35AL4kbz0N216PjAIbBcs">

The latest version of vulkanmod now rund under xwayland by default. So this shouldn't happen anymore. Can you conform? If it works then this can be closed
